src/main/java/io/confluent/idesidecar/restapi/connections/DirectConnectionState.java

```diff
@@ -48,5 +78,126 @@ public Optional<Credentials> getSchemaRegistryCredentials(String clusterId) {
     return Optional.ofNullable(credentials);
   }
 
-  // TODO: DIRECT connections need validation checks
+  @Override
+  public Future<ConnectionStatus> getConnectionStatus() {
+    return Future.join(
+        getKafkaConnectionStatus(),
+        getSchemaRegistryConnectionStatus()
+    ).map(cf -> {
+      var futures = cf.list();
+      var kafkaStatus = (KafkaClusterStatus) futures.get(0);
+      var srStatus = (SchemaRegistryStatus) futures.get(1);
+      return ConnectionStatusBuilder
+          .builder()
+          .kafkaCluster(kafkaStatus)
+          .schemaRegistry(srStatus)
+          .build();
+    });
+  }
+
+  protected Future<KafkaClusterStatus> getKafkaConnectionStatus() {
+    var kafkaConfig = spec.kafkaClusterConfig();
+    if (kafkaConfig == null) {
+      return Future.succeededFuture(
+          ConnectionStatusKafkaClusterStatusBuilder
+              .builder()
+              .state(ConnectedState.NONE)
+              .build()
+      );
+    }
+    // There is a Kafka configuration, so validate the connection by creating an AdminClient
+    // and describing the cluster.
+    try (var adminClient = createAdminClient(kafkaConfig)) {
+      var clusterDesc = adminClient.describeCluster();
+      var actualClusterId = clusterDesc.clusterId().get(5, TimeUnit.SECONDS);
+      // Set the cluster ID
+      getSpec().kafkaClusterConfig().withId(actualClusterId);
+      return Future.succeededFuture(
+          ConnectionStatusKafkaClusterStatusBuilder
+              .builder()
+              .state(ConnectedState.SUCCESS)
+              .build()
+      );
+    } catch (Exception e) {
+      // The connection failed, so successfully return the failed state
+      return Future.succeededFuture(
+          ConnectionStatusKafkaClusterStatusBuilder
+              .builder()
+              .state(ConnectedState.FAILED)
+              .errors(
+                  new AuthErrors().withSignIn(
```

Comment:
The whole point of #162 is to deprecate and move away from the `status.authentication` object, since it only applies to CCloud. The CCloud-related status is available under a nested `status.ccloud` object (structurally identical to `status.authentication`), while direct connection errors (incl auth-related errors) are recorded under `status.kafka_cluster` or `status.schema_registry` (depending upon where the error is).